syntax="proto3";


package grpc;

service ChatMessages {
<<<<<<< HEAD
    rpc GetNewMessages(User) returns (ChatMessage);
    rpc SendNewMessages(ChatMessage) returns (RequestSuccess);
=======
    rpc ChatStream(User) returns (stream ChatMessage);
    rpc SendMessage(ChatMessage) returns (RequestSuccess);
>>>>>>> 5c56dbb (updated protobuf and generated code)
}

message User {
    string userID = 1;
    string userName = 2;
}

message ChatMessage {
    string senderID = 1;
    string recipientID = 2;
    string messageText = 3;
}

message RequestSuccess {
    bool receivedRequest = 1;
}

message Empty { }